pkg/resourceapply/apps.go

```diff
@@ -17,7 +17,39 @@ func ApplyStatefulSetWithControl(
 	required *appsv1.StatefulSet,
 	options ApplyOptions,
 ) (*appsv1.StatefulSet, bool, error) {
-	return ApplyGeneric[*appsv1.StatefulSet](ctx, control, recorder, required, options)
+	return ApplyGenericWithHandlers[*appsv1.StatefulSet](
+		ctx,
+		control,
+		recorder,
+		required,
+		options,
+		nil,
+		func(required *appsv1.StatefulSet, existing *appsv1.StatefulSet) string {
```

Comment:
the selector recreate is only graceful, if done with orphan delete, so i think we need it enforce at this or lower level, not the caller that can forget to set it.

Reply:
could the propagation policy be a return value here?

Reply:
done